packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -91,139 +123,600 @@ const propTypes = {
+      lastInputRefCallback(node);
     }
-  }
+  }, [dateFormatOrder, lastInputRefCallback]);
 
-  handleOnChange(event) {
-    if (!DateUtil.validDateInput(event.target.value)) {
-      return;
+  // Triggers the onClick callback to launch the dropdown picker for the scenario when the default date is invalid and
+  // the calendar button is clicked which should clear the value and launch the dropdown picker
+  useEffect(() => {
+    if (shouldShowPicker && onClick) {
+      onClick();
     }
+  }, [shouldShowPicker, onClick]);
 
-    if (this.props.onChange) {
-      this.props.onChange(event);
+  // Sets the date state based on the passed in value prop, or if it changes via a calendar click.
+  useEffect(() => {
+    if (DateUtil.isValidDate(value, momentDateFormat)) {
+      const parseDate = DateUtil.getDateInputValues(DateUtil.dateOrder.YMD, dateValue, '-');
+      dateDispatch(parseDate);
+    } else if (dateValue === '') {
+      dateDispatch({ day: '', month: '', year: '' });
     }
-  }
+  }, [momentDateFormat, dateValue, value]);
 
-  handleOnKeyDown(event) {
-    if (this.props.onKeyDown) {
-      this.props.onKeyDown(event);
+  /**
+   * Moves focus to the correct input depending on date ordering. Focus changing is
+   * disabled if a complete date has been entered in order to make single input
+   * corrections easier, and is re-enabled if the whole date is erased.
+   * @param {string} inputValue - The value from the current input.
+   * @param {number} type - The input type, based on DateUtil.inputType.
+   */
+  const moveFocusOnChange = (inputValue, type) => {
+    if (dateFormatOrder === DateUtil.dateOrder.MDY) {
+      if (inputValue.length === 2) {
+        if (type === DateUtil.inputType.MONTH) {
+          dayInputRef.current.focus();
+        } else {
+          yearInputRef.current.focus();
+        }
+      }
+    } else if (dateFormatOrder === DateUtil.dateOrder.DMY) {
+      if (inputValue.length === 2) {
+        if (type === DateUtil.inputType.DAY) {
+          monthInputRef.current.focus();
+        } else {
+          yearInputRef.current.focus();
+        }
+      }
+    } else if (dateFormatOrder === DateUtil.dateOrder.YMD) {
+      if (inputValue.length === 4) {
+        monthInputRef.current.focus();
+      } else if (inputValue.length === 2 && type === DateUtil.inputType.MONTH) {
+        dayInputRef.current.focus();
+      }
+    }
+  };
+
+  /**
+   * Sets state for the day, month, and year.
+   * @param {object} event - Event object
+   * @param {string} inputValue - The input value to set in state.
+   * @param {number} type - The inputType (day, month, or year).
+   */
+  const setDate = (event, inputValue, type) => {
+    if (type === DateUtil.inputType.DAY) {
+      dateDispatch({ day: inputValue });
+    } else if (type === DateUtil.inputType.MONTH) {
+      dateDispatch({ month: inputValue });
+    } else {
+      dateDispatch({ year: inputValue });
+    }
+
+    if (event.type !== 'keydown') {
```

Comment:
can we move this sting to constant ?

Reply:
Since, 'keydown' is being used only here and is not being reused, I decided to use it as is here without declaring a constant.

Reply:
as part of coding standard it is always better to declare it as the constant instead of hardcoding it. and also if we declare this as a constant now. it can be re used in the future as well.  if we hardocde the string there is a high chance of using the same pattern in the future also.

Reply:
changed here : https://github.com/cerner/terra-framework/pull/1208/commits/c7acad33dc268eabd63e7962ef688604de72e8bc